newton/tests/test_ik.py

```diff
@@ -0,0 +1,684 @@
+        ee_offset = wp.vec3(1.0, 0.0, 0.0)  # 1m from joint to tip
+        num_links = 2  # 2 links per articulation
+        coords_per_env = 2  # 2 revolute joints per articulation
+        total_residuals = 3  # Position only (x, y, z)
+
+        # Create different target positions for each environment
+        target_positions = []
+        for env_idx in range(num_envs):
+            x_offset = env_idx * 3.0
+            target_positions.append(
+                [
+                    x_offset + 1.5 + env_idx * 0.1,  # x
+                    0.5 + env_idx * 0.1,  # y
+                    0.0,  # z
+                ]
+            )
+
+        target_position_array = wp.array(target_positions, dtype=wp.vec3)
+
+        # Create position objective
+        objective = ik.PositionObjective(
+            link_index=ee_link_index,
+            link_offset=ee_offset,
+            target_positions=target_position_array,
+            num_links=num_links,
+            num_envs=num_envs,
+            total_residuals=total_residuals,
+            residual_offset=0,
+        )
+
+        # Override supports_analytic to return True for testing
+        objective.supports_analytic = lambda: True
+
+        # Allocate arrays
+        residuals = wp.zeros((num_envs, total_residuals), dtype=wp.float32, requires_grad=True)
+        jacobian_autodiff = wp.zeros((num_envs, total_residuals, coords_per_env), dtype=wp.float32)
+        jacobian_analytic = wp.zeros((num_envs, total_residuals, coords_per_env), dtype=wp.float32)
+
+        # Compute residuals
+        objective.compute_residuals(state, model, residuals, 0)
+
+        # Compute Jacobian with autodiff
+        tape = wp.Tape()
+        with tape:
+            # Need fresh state computation in tape
+            newton.core.articulation.eval_fk(model, model.joint_q, model.joint_qd, state, None)
+            residuals_compute = wp.zeros((num_envs, total_residuals), dtype=wp.float32, requires_grad=True)
+            objective.compute_residuals(state, model, residuals_compute, 0)
+            residuals_flat = residuals_compute.flatten()
+
+        tape.outputs = [residuals_flat]
+        objective.compute_jacobian_autodiff(tape, model, jacobian_autodiff, 0)
+
+        # Compute Jacobian analytically
+        objective.compute_jacobian_analytic(state, model, jacobian_analytic, 0)
+
+        # Compare
+        J_auto = jacobian_autodiff.numpy()
+        J_analytic = jacobian_analytic.numpy()
+
+        # Check each environment
+        for env_idx in range(num_envs):
+            # Get actual EE position for debugging
+            body_idx = env_idx * num_links + ee_link_index
+            body_tf = state.body_q.numpy()[body_idx]
+
+            # Convert properly to avoid warnings
+            pos = wp.vec3(body_tf[0], body_tf[1], body_tf[2])
+            rot = wp.quat(body_tf[3], body_tf[4], body_tf[5], body_tf[6])
+            tf = wp.transform(pos, rot)
+            wp.transform_point(tf, ee_offset)
```

Comment:
Here you just compare the Jacobians, I don't think you need these previous computations?